### ISSUE TITLE
-Wbraced-scalar-init warns inappropriately

### ISSUE THREAD

**Issue by Leengit** [llvm/llvm-project#54702]
Braces around the returned value serve a legitimate purpose in
```c++
char f()
{
  int i = 65537;
  return{ i };
}
```
They tell the compiler to fail because the return value is narrowed; without the braces this would compile.  Yet, `-Wbraced-scalar-init` warns that this is a superfluous use of braces.  Likewise if the return value of `f` is instead `int` the braces legitimately ask the compiler to check for narrowing, which in this case is not occurring; in this situation too, `-Wbraced-scalar-init` should not be warning that this is a superfluous use of braces.  (In practical cases, the returned value might be from a function that has a return type that is not so readily apparent and the code writer should instead be encouraged to use braces in situations where narrowing is a concern!)
